numba/targets/listobj.py

```diff
@@ -408,6 +499,7 @@ def getitem_list(context, builder, sig, args):
     index = args[1]
 
     index = inst.fix_index(index)
+    inst.guard_index(index, msg="getitem out of range")
```

Comment:
Just had a look at `guard_index` which calls into `is_out_of_bounds`, seems like the underflow condition is triggered by a negative index? Should this not be valid so long as the magnitude is `<= len(list)`? If this is the case, it doesn't need fixing here and can be done independently.

Reply:
the negative index can be `<= -size`.  `fix_index` only adds the `size` to it.  the result can still be negative.